qt/python/mantidqt/utils/asynchronous.py

```diff
@@ -90,6 +90,24 @@ def abort(self):
         time.sleep(0.1)
 
 
+class Receiver(object):
+    output, exc_value = None, None
```

Comment:
This is actually a class level variable, if two tasks finish at the same time A will override B's copy
Instead we should move these attributes to `__init__` so that they live with the current instance

Reply:
Thanks for spotting that!

Reply:
I think for clarity this is a good idea but just for reference the sharing is only an issue for [mutable types](https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables), basically the same rules as for parameters in functions/methods with default values.